### ISSUE TITLE
Error while retrieving a payout batch

### ISSUE THREAD

**Issue by AlexanderKarpanProdege** [bitpay/php-bitpay-client-v2#238]
We use PHP 8.2 and `bitpay/sdk:7.3` (the same error is on `7.2` and `dev-master`)

A call to `GET /payoutBatches/:payoutBatchId` throws an exception: 
`Cannot use object of type stdClass as array`
Stack trace:
```
vendor/bitpay/sdk/src/BitPaySDK/Model/Payout/PayoutBatch.php:87
vendor/bitpay/sdk/src/BitPaySDK/Model/Payout/PayoutBatch.php:243
???:0
vendor/bitpay/sdk/src/BitPaySDK/Util/JsonMapper/JsonMapper.php:651
vendor/bitpay/sdk/src/BitPaySDK/Util/JsonMapper/JsonMapper.php:249
vendor/bitpay/sdk/src/BitPaySDK/Client.php:2030
```

The JSON response from the endpoint seems to be valid. When the SDK tries to deserialize it with `JsonMapper` it fails on array of 'instructions'.

vendor/bitpay/sdk/src/BitPaySDK/Model/Payout/PayoutBatch.php:79-92
```
    private function _computeAndSetAmount()
    {
        $amount = 0.0;
        if ($this->_instructions) {
            foreach ($this->_instructions as $instruction) {
                if ($instruction instanceof PayoutInstruction) {
                    $amount += $instruction->getAmount();
                } else {
                    $amount += $instruction['amount']; // <-- This line
                }
            }
        }
        $this->_amount = $amount;
    }
```

Instructions are deserialized as an array of `stdClass`, so it fails on the line ` $amount += $instruction['amount'];` which used to be `$amount += $instruction->amount;` in the previous versions.

**Comment by bobbrodie** [comment]
Hi @AlexanderKarpanProdege - we've been discussing this internally, and here's where we landed:

- We're going to test Payout Batches with the previous line (it was intentionally updated at one point) and as long as it works, we'll release that as 7.3.1 this week
- Payout Batches are deprecated and replaced with Payout Groups, which we'll add to both 7.4.0 and the upcoming 8.0.0 release

**Comment by AlexanderKarpanProdege** [comment]
Hi @bobbrodie !
Thank you for the reply and for the heads-up.
Looking forward for the fix.

I believe we are ready to implement Payout Groups as soon as it is available.

**Comment by bobbrodie** [comment]
Hey there @AlexanderKarpanProdege , just a quick heads up - we merged a PR into `master` with the following change:

```php
    private function _computeAndSetAmount()
    {
        $amount = 0.0;
        if ($this->_instructions) {
            foreach ($this->_instructions as $instruction) {
                if ($instruction instanceof PayoutInstruction) {
                    $amount += $instruction->getAmount();
                } elseif ($instruction instanceof \stdClass) {
                    $amount += $instruction->amount;
                } else {
                    $amount += $instruction['amount'];
                }
            }
        }
        $this->_amount = $amount;
    }
```

The previous change was to address a case where it needed to be treated as an array. I'll have to double check, but it may have been related to earlier versions of `JsonMapper`, so we're now checking if it's `\stdClass` and handling it in the previous way.

I'll keep this open while we prepare a release, but if you are able to test against `master-dev` and let us know if you have any other issues that would be awesome.

**Comment by bobbrodie** [comment]
Released version 7.3.1 - if you have any issues please let us know. Thanks!

**Comment by AlexanderKarpanProdege** [comment]
Everything works! Thank you, @bobbrodie !
Looking forward for Payout Groups!
